_overviews/scala3-book/taste-modeling.md

```diff
@@ -50,27 +67,57 @@ trait Runner:
   def startRunning(): Unit = println("I’m running")
   def stopRunning(): Unit = println("Stopped running")
 ```
+{% endtab %}
+{% endtabs %}
 
 Given those traits, here’s a `Dog` class that extends all of those traits while providing a behavior for the abstract `speak` method:
 
+{% tabs traits-class class=tabs-scala-version %}
+{% tab 'Scala 2' for=traits-class %}
+```scala
+class Dog(name: String) extends Speaker, TailWagger, Runner {
+  def speak(): String = "Woof!"
+}
+```
+{% endtab %}
+
+{% tab 'Scala 3' for=traits-class %}
 ```scala
 class Dog(name: String) extends Speaker, TailWagger, Runner:
   def speak(): String = "Woof!"
 ```
+{% endtab %}
+{% endtabs %}
 
 Notice how the class extends the traits with the `extends` keyword.
 
 Similarly, here’s a `Cat` class that implements those same traits while also overriding two of the concrete methods it inherits:
 
+{% tabs traits-override class=tabs-scala-version %}
+{% tab 'Scala 2' for=traits-override %}
+```scala
+class Cat(name: String) extends Speaker, TailWagger, Runner {
+  def speak(): String = "Meow"
+  override def startRunning(): Unit = println("Yeah ... I don’t run")
+  override def stopRunning(): Unit = println("No need to stop")
+}
+```
+{% endtab %}
+
+{% tab 'Scala 3' for=traits-override %}
 ```scala
 class Cat(name: String) extends Speaker, TailWagger, Runner:
   def speak(): String = "Meow"
   override def startRunning(): Unit = println("Yeah ... I don’t run")
   override def stopRunning(): Unit = println("No need to stop")
 ```
+{% endtab %}
+{% endtabs %}
 
 These examples show how those classes are used:
 
+{% tabs traits-use class=tabs-scala-version %}
+{% tab 'Scala 2 and 3' for=traits-use %}
```

Comment:
Cat and Dog are not case classes, so this syntax is not valid for Scala 2

Reply:
changed.